sbi/inference/snpe/snpe_base.py

```diff
@@ -158,18 +161,26 @@ def __call__(
         for round_, num_sims in enumerate(num_sims_per_round):
 
             # Run simulations for the round.
-            theta, x, prior_mask = self._run_simulations(round_, num_sims)
-            x_shape = x_shape_from_simulation(x)
+            theta, x = self._run_simulations(round_, num_sims)
+
+            # Append data to self._theta_bank, self._x_bank, and self._prior_masks.
+            self._append_training_data(
```

Comment:
maybe remove the comment and change the function signature to something like `_append_to_banks(theta, x, external_data, round_, exclude_invalid_x)`